### PR TITLE
chore(data)!: breaking data changes for future compatibility

### PR DESCRIPTION
**Pull request by IceTDrinker**
- invert the LweKeyswitchKey level order and propagate change
- remove dependency on unsupported wopbs keys for the HL keys

### CODE REVIEW

File: tfhe/src/high_level_api/backward_compatibility/keys.rs [hunk 1]
```diff
@@ -184,42 +189,25 @@ pub(crate) enum IntegerConfigVersions {
...
     fn upgrade(self) -> Result<IntegerClientKeyV2, Self::Error> {
-        Ok(IntegerClientKeyV2 {
-            key: self.key,
-            dedicated_compact_private_key: self.dedicated_compact_private_key,
-            compression_key: self.compression_key,
-        })
+        Err(crate::Error::new(
+            "Unable to load IntegerClientKey, \
+            this format is unsupported by this TFHE-rs version."
+                .to_string(),
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/backward_compatibility/keys.rs:209]: improve err message with the diff types

File: tfhe/src/high_level_api/backward_compatibility/keys.rs [hunk 2]
```diff
@@ -247,52 +235,39 @@ impl Upgrade<IntegerClientKey> for IntegerClientKeyV2 {
...
-            key: self.key,
-            wopbs_key: self.wopbs_key,
-            cpk_key_switching_key_material: None,
-            compression_key: None,
-            decompression_key: None,
-        })
+    fn upgrade(self) -> Result<UnsupportedIntegerServerKeyV1, Self::Error> {
+        Err(crate::Error::new(
+            "Unable to load IntegerServerKey, \
+            this format is unsupported by this TFHE-rs version."
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/backward_compatibility/keys.rs:256]: improve err message with the diff types

File: tfhe/src/shortint/backward_compatibility/key_switching_key.rs [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+use tfhe_versionable::{Upgrade, Version, VersionsDispatch};
 
 use crate::shortint::key_switching_key::{
     CompressedKeySwitchingKeyMaterial, KeySwitchingKeyMaterial,
 };
 use crate::shortint::{CompressedKeySwitchingKey, KeySwitchingKey};
 
+#[derive(Version)]
+pub struct UnsupportedCompressedKeySwitchingKeyMaterialV0;
+
```
Comment (IceTDrinker) [inline tfhe/src/shortint/backward_compatibility/key_switching_key.rs:10]: move type lower in the file

**Comment by IceTDrinker** [comment]
@guillermo-oyarzun if you can check the cuda code: we have inverted the levels of the keys for the keyswitch key and packing keyswitch key in our rust code to be consistent with the rest of the code base, normally the diff on the cuda files is small

@mayeul-zama if you can check the core crypto part for consistency and that I did not miss anything

@nsarlin-zama you already know but I would like your expert optinion on the data management

**Comment by guillermo-oyarzun** [comment]
GPU changes seem fine, I am just running some benches to see if affects performance (I don't think so)

**Comment by IceTDrinker** [comment]
> GPU changes seem fine, I am just running some benches to see if affects performance (I don't think so)

worst case @guillermo-oyarzun you can invert the levels on the GPU side and when loading keys from the CPU you can invert them during the memory transfer, I just made the implems match today

did you see any impact on performance ?

**Review by guillermo-oyarzun — APPROVED** [review]

**Comment by guillermo-oyarzun** [comment]
> > GPU changes seem fine, I am just running some benches to see if affects performance (I don't think so)
> 
> worst case @guillermo-oyarzun you can invert the levels on the GPU side and when loading keys from the CPU you can invert them during the memory transfer, I just made the implems match today
> 
> did you see any impact on performance ?

No changes in performance, so go for it

File: tfhe/src/core_crypto/backward_compatibility/entities/seeded_lwe_keyswitch_key.rs [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+
+impl<Scalar: UnsignedInteger, C: Container<Element = Scalar>> Upgrade<SeededLweKeyswitchKey<C>>
+    for UnsupportedSeededLweKeyswitchKeyV0
+{
+    type Error = crate::Error;
+
+    fn upgrade(self) -> Result<SeededLweKeyswitchKey<C>, Self::Error> {
+        Err(crate::Error::new(
+            "Unable to load SeededLweKeyswitchKey, \
+            this format is UnsupportedSeededLweKeyswitchKeyV0 by this TFHE-rs version."
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/core_crypto/backward_compatibility/entities/seeded_lwe_keyswitch_key.rs:16]: Indeed, search and replace failed me lol

**Comment by IceTDrinker** [comment]
> Maybe we should also completely remove the CompactFheXXX types since they are not even tested anymore ?

While I understand the reasoning, it should not be broken, so :/

**Comment by nsarlin-zama** [comment]
> > Maybe we should also completely remove the CompactFheXXX types since they are not even tested anymore ?
> 
> While I understand the reasoning, it should not be broken, so :/

I was thinking of the types of `high_level_api/backward_compatiblity` that were already flagged as deprecated since 0.7:

`CompactFheBool`, `CompactFheBoolList`, `CompactFheIntList`, `CompactFheInt`, `CompactFheUintList` and `CompactFheUint`

**Review by nsarlin-zama — APPROVED** [review]
Data changes look good to me that way !

**Pull request merged**